### DISCUSSION TITLE
Help Installing in Google Colab

### DISCUSSION THREAD

**Discussion by hodgesz** [WeOpenML/PandaLM#25] [Q&A] [answered]
First off, thanks for building this project. It is something really needed in the community to evaluate LLMs and chatbots.

When I try pip install -r requirements.txt I get the following error.

Requirement already satisfied: triton==2.0.0 in /usr/local/lib/python3.10/dist-packages (from torch==2.0.0->-r requirements.txt (line 13)) (2.0.0)
INFO: pip is looking at multiple versions of torchvision to determine which version is compatible with other requirements. This could take a while.
ERROR: Cannot install -r requirements.txt (line 1), -r requirements.txt (line 14), -r requirements.txt (line 15), -r requirements.txt (line 9) and torch==2.0.0 because these package versions have conflicting dependencies.

The conflict is caused by:
    The user requested torch==2.0.0
    accelerate 0.18.0 depends on torch>=1.4.0
    peft 0.2.0 depends on torch>=1.13.0
    torchaudio 2.0.0 depends on torch==2.0.0
    torchvision 0.15.0 depends on torch==2.0.0+cu117

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts

Any ideas?

Or is there a way to run on my Mac M2 Max GPU like the following?

Metal Backend (Mac Computers with Apple Silicon or AMD GPUs)
Use --device mps to enable GPU acceleration on Mac computers (requires torch >= 2.0). Use --load-8bit to turn on 8-bit compression.

python3 -m fastchat.serve.cli --model-path lmsys/vicuna-7b-v1.3 --device mps --load-8bit

**Comment by zhuohaoyu** [comment]
Hi, for reproducing exact results from our experiments, we gave exact version numbers for packages to recreate our exact environment and the requirements are intended for a clean environment. For your use case, you could simply use a existing PyTorch 2.0 environment from colab, or follow https://pytorch.org/get-started/locally/ for your own platform.

**Comment by hodgesz** [comment, marked as answer]
I appreciate your quick reply and figured it was something I was doing wrong as I can install in my Mac with no issues, but having this problem in Colab, which must mean not a clean Python environment. Since my Mac is limited without CUDA, I was trying to find an alternative in Colab, but brand new to the tool. I will work on adding virtual environment like Conda to Colab so I can better control the dependencies. Thanks again!
